### DISCUSSION TITLE
How do I insert a <script> tag into the HTML document's head for just one component?

### DISCUSSION THREAD

**Discussion by moonman239** [vercel/next.js#57541] [Help]
### Summary

I'm trying to use a Stripe pricing table in a component.
To do this, I have to insert the following line in the document head (between <head> and </head>.)
```
<script async src="https://js.stripe.com/v3/pricing-table.js"></script>
```
In a React app, this would be easy - find index.html and add the above line to its head.

However, this is not possible in NextJS.

I tried returning this in the component:

```
return <>
            <Script async strategy="beforeInteractive" src="https://js.stripe.com/v3/pricing-table.js"></Script>
            <stripe-pricing-table pricing-table-id={pricingTableId} publishable-key={publishableKey} client-reference-id={facebookUserID} />
    </>
```

but I get a warning that says I shouldn't use "beforeInteractive" outside of the "pages/_document.js" file.

However, I am using the App Router, so there is no "pages" folder.

Putting that line in between the <html> and </html> tags in the return statement for layout.tsx produces the same warning, and a run-time error that says hydration failed due to the UI not being rendered as expected or something.

How do I have NextJS put my <script> tag where it needs to be?

**Comment by icyJoseph** [comment]
I haven't tried this myself... but you could use the `useServerInsertedHTML` hook from `next/navigation`. Take a look here:

- https://github.com/t3dotgg/cursed-next-patterns/blob/main/src/app/globalThis/helpers/embed-on-server.tsx

Would you mind giving it a go? And see where it takes you?

**Reply by moonman239** [reply]
Sure, what does useServerInsertedHTML do?

**Reply by icyJoseph** [reply]
Typically you use it to inject style tags into the head of the document.

**Comment by moonman239** [comment]
So, this code below puts the script tag in the body and not the head, but that doesn't seem to affect the functionality as long as the script gets loaded before the <stripe-pricing-table> tag.

Here's my code:

```
import Script from "next/script";

type PricingTableProps = {
    id: string,
    clientReferenceId: string;
    publishableKey: string
}; 

export default function PricingTable(props: PricingTableProps)
{
    return <>
    <Script src="https://js.stripe.com/v3/pricing-table.js"></Script>
    <stripe-pricing-table pricing-table-id={props.id} publishable-key={props.publishableKey} client-reference-id={props.clientReferenceId} />
    </>
}
```
